### ISSUE TITLE
get_status() is broken in 3D when the layers do not have the same number of dimensions

### ISSUE THREAD

**Issue by kevinyamauchi** [napari/napari#3118]
## 🐛 Bug
When mousing around the canvas in 3D mode when the layers have different numbers of dimensions, if the layer with fewer dimensions is selected, we get an `IndexError`. I am pretty sure this is because the layer doesn't have as many dimensions and we aren't mapping `dims_displayed` properly. To be a bit more concrete:

If `layer_0` is 5d and `layer_1` is 3d. In 3D display mode, by default, the dims displayed values will be:
- `viewer.dims.displayed`: [2, 3, 4]
- `layer_0._dims_displayed`: [2, 3, 4]
- `layer_1.dims_displayed`: [0, 1, 2]

We use the `viewer.dims.displayed` value (`[2, 3, 4]`) in the mouse event, so there is an indexing error when that is applied to `layer_1`, which only has 3 dimensions.

Thus, I think we need to add a line to the section below that converts the world dims_displayed to layer dims_displayed. Do the layers already have such a method that we should use (cc @sofroniewn )?

https://github.com/napari/napari/blob/871a81f10763a19e7308dd99cd57095eb16bf929/napari/layers/base/base.py#L1013-L1014

## To Reproduce

Run the script below and then mouse around the canvas
```python
import napari
import numpy as np

data_4d = np.random.random((5, 5, 20, 20))
viewer = napari.view_image(data_4d)

data_3d =  np.random.random((5, 20, 20))
viewer.add_image(data_3d)
viewer.dims.ndisplay = 3
napari.run()
```

Traceback:
```python
~/Documents/napari/napari/layers/base/base.py in get_ray_intersections(self=<Image layer 'data_3d'>, position=(0.6302527216009963, 0.0, 9.5), view_direction=(0.0, 0.0, 1.0), dims_displayed=[3, 0, 1], world=False)
   1208             dims_displayed = dims_displayed
   1209             dims_displayed_mask = np.zeros_like(position, dtype=bool)
-> 1210             dims_displayed_mask[dims_displayed] = True
        dims_displayed_mask = <class 'numpy.ndarray'> (3,) bool
        dims_displayed = [3, 0, 1]
   1211 
```

## Expected behavior
`dims_displayed` should be converted to the layer `dims_displayed`.

## Environment
napari: 0.4.11.dev94+geeab5ca2.d20210729
Platform: Darwin-19.6.0-x86_64-i386-64bit
System: MacOS 10.15.7
Python: 3.7.9 (default, Aug 31 2020, 07:22:35) [Clang 10.0.0 ]
Qt: 5.15.2
PyQt5: 5.15.2
NumPy: 1.21.1
SciPy: 1.6.0
Dask: 2020.12.0
VisPy: 0.6.6

OpenGL:
- GL version: 2.1 INTEL-14.7.17
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 2560x1440, scale 1.0
- screen 2: resolution 1440x900, scale 2.0

Plugins:
- console: 0.0.3
- napari-properties-viewer: 0.0.1
- scikit-image: 0.4.11.dev94+geeab5ca2.d20210729
- svg: 0.1.4

## Additional context
Bug introduced in #3074

**Comment by sofroniewn** [comment]
>Thus, I think we need to add a line to the section below that converts the world dims_displayed to layer dims_displayed. Do the layers already have such a method that we should use (cc @sofroniewn )?

there was code for doing this in the code base before, but i think it might be gone now. i think now best to add a new util method, long term though i really do want us to move away from ever needing `layer._dims_displayed

**Comment by kevinyamauchi** [comment]
Thanks @sofroniewn ! I agree we should avoid adding new usages of `layer._dims_displayed`. As suggested, I made a small util function that converts world `dims_displayed_world` to  `dims_displayed_layer` in #3121 .

**Comment by kevinyamauchi** [comment]
Closed by #3121
